test/integration/tlsroute_test.go

```diff
@@ -356,11 +463,293 @@ func TestTLSRouteEssentials(t *testing.T) {
 }
 
+// TestTLSRouteReferencePolicy tests cross-namespace certificate references. These are technically implemented within
+// Gateway Listeners, but require an attached Route to see the associated certificate behavior on the proxy
+func TestTLSRouteReferencePolicy(t *testing.T) {
+	backendPort := gatewayv1alpha2.PortNumber(tcpEchoPort)
+	t.Log("locking TLS port")
+	tlsMutex.Lock()
+	defer func() {
+		t.Log("unlocking TLS port")
+		tlsMutex.Unlock()
+	}()
+	ns, cleanup := namespace(t)
+	defer cleanup()
+
+	otherNs, err := clusters.GenerateNamespace(ctx, env.Cluster(), t.Name())
+	require.NoError(t, err)
+
+	// TODO consolidate into suite and use for all GW tests?
+	// https://github.com/Kong/kubernetes-ingress-controller/issues/2461
+	t.Log("deploying a supported gatewayclass to the test cluster")
+	c, err := gatewayclient.NewForConfig(env.Cluster().Config())
+	require.NoError(t, err)
+	gwc := &gatewayv1alpha2.GatewayClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: uuid.NewString(),
+		},
+		Spec: gatewayv1alpha2.GatewayClassSpec{
+			ControllerName: gateway.ControllerName,
+		},
+	}
+	gwc, err = c.GatewayV1alpha2().GatewayClasses().Create(ctx, gwc, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	defer func() {
+		t.Log("cleaning up gatewayclasses")
+		if err := c.GatewayV1alpha2().GatewayClasses().Delete(ctx, gwc.Name, metav1.DeleteOptions{}); err != nil {
+			if !apierrors.IsNotFound(err) {
+				assert.NoError(t, err)
+			}
+		}
+	}()
+
+	t.Log("configuring secrets")
+	secrets := []*corev1.Secret{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("7428fb98-180b-4702-a91f-61351a33c6e8"),
+				Name:      "secret1",
+				Namespace: ns.Name,
+			},
+			Data: map[string][]byte{
+				"tls.crt": []byte(tlsRouteTLSPairs[0].Cert),
+				"tls.key": []byte(tlsRouteTLSPairs[0].Key),
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:  types.UID("7428fb98-180b-4702-a91f-61351a33c6e9"),
+				Name: "secret2",
+			},
+			Data: map[string][]byte{
+				"tls.crt": []byte(tlsRouteTLSPairs[1].Cert),
+				"tls.key": []byte(tlsRouteTLSPairs[1].Key),
+			},
+		},
+	}
+
+	t.Log("deploying secrets")
+	secret1, err := env.Cluster().Client().CoreV1().Secrets(ns.Name).Create(ctx, secrets[0], metav1.CreateOptions{})
+	assert.NoError(t, err)
+	secret2, err := env.Cluster().Client().CoreV1().Secrets(otherNs.Name).Create(ctx, secrets[1], metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	defer func() {
+		t.Logf("ensuring that Secret %s is cleaned up", secret1.Name)
+		if err := env.Cluster().Client().CoreV1().Secrets(ns.Name).Delete(ctx, secret1.Name, metav1.DeleteOptions{}); err != nil {
+			if !apierrors.IsNotFound(err) {
+				require.NoError(t, err)
+			}
+		}
+		t.Logf("ensuring that Secret %s is cleaned up", secret2.Name)
+		if err := env.Cluster().Client().CoreV1().Secrets(otherNs.Name).Delete(ctx, secret2.Name, metav1.DeleteOptions{}); err != nil {
+			if !apierrors.IsNotFound(err) {
+				require.NoError(t, err)
+			}
+		}
+	}()
```

Comment:
Similar to a previous comment, with KTF's new `cluster.Cleaner` this whole block could become:

```golang
	cleaner.Add(secret1)
	cleaner.Add(secret2)
```